### ISSUE TITLE
Horrible GAT code that's accepted by nightly fails to parse

### ISSUE THREAD

**Issue by casey** [dtolnay/syn#973]
I'm experimenting with a derive macro that uses GATs, and some particularly heinous code which is accepted by nightly is rejected by `syn`:

```rust
#![feature(generic_associated_types)]
#![allow(incomplete_features)]

trait Bar {
  type Baz<B>;
}

fn foo<A, B, C>()
  where A: Bar<Baz<B> = C>,
{
}
```

The error message is ``expected `,` ``, and I believe it's the `Baz<B>` in the `where` clause that's the culprit.

I wrote a test which fails that looks like this:

```rust
#[test]
fn parse() {
  let tokens = quote::quote!(
    trait Bar {
      type Baz<B>;
    }

    fn foo<A, B, C>()
      where A: Bar<Baz<B> = C>,
    {
    }
  );

  syn::parse2::<syn::File>(tokens).unwrap();
}
```

I wish I could forget I ever wrote such a thing as `where A: Bar<Baz<B> = C>` and just go on with my life, but I actually need it to get my derive macro working.

I wonder if this is perhaps a policy issue? (I.e. that `syn` doesn't necessarily support all bleeding edge syntax accepted by nightly.)

PS Thank you for this indispensable crate!

**Comment by dtolnay** [comment]
Fixed in 1.0.66.

**Comment by casey** [comment]
Nice!
